### PR TITLE
Add Autoscaler docs and support cli and appfile

### PR DESCRIPTION
**Pull request by zzxwill**
added Autoscaler docs and add spec.extension.template
for cli and appfile

### CODE REVIEW

File: docs/developers/references/traits/autoscale.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Automatically scale workload by cron or resource utilization
+
+## Prerequisites
+ - [ ] [KEDA v2.0 Beta](https://keda.sh/blog/keda-2.0-beta/)
+ 
+   KEDA will be automatically deployed during veal installation, so just run the following command.
```
Comment (wonderflow) [inline docs/developers/references/traits/autoscale.md:6]:
```suggestion
   KEDA will be automatically deployed during vela installation, so just run the following command.
```

File: charts/vela-core/templates/defwithtemplate/autoscale.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                  }
+              }, ...]
+          }
+        }
+      parameter: {
+      	minReplicas: *1 | int
+      	maxReplicas: *4 | int
+      	name:        *"" | string
+      	type:        *"cron" | string
+      	startAt:     *"00:01" | string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/autoscale.yaml:44]: this should be required

```suggestion
      	startAt:     string
```

File: charts/vela-core/templates/defwithtemplate/autoscale.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      	duration:    *"" | string
+      	days:        *"Monday, Tuesday" | string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/autoscale.yaml:46]: this should be required

```suggestion
      	days:      string
```

File: charts/vela-core/templates/defwithtemplate/autoscale.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              }, ...]
+          }
+        }
+      parameter: {
+      	minReplicas: *1 | int
+      	maxReplicas: *4 | int
+      	name:        *"" | string
+      	type:        *"cron" | string
+      	startAt:     *"00:01" | string
+      	duration:    *"" | string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/autoscale.yaml:45]:
```suggestion
      	duration:  string
```

This should be required

File: docs/developers/references/traits/autoscale.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+   ```shell
+   $ vela install
+   ```
+   
+   You can also double check whether KEDA is successfully installed.
+   ```
+   $ helm list -n keda
+   NAME	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART             	APP VERSION
+   keda	keda     	1       	2020-11-02 17:01:16.122247 +0800 CST	deployed	keda-2.0.0-beta1.2	2.0.0-beta
+   ```
```
Comment (wonderflow) [inline docs/developers/references/traits/autoscale.md:16]: please don't expose so much details for developers. I think we should remove this section.

Or we should using `vela traits` to provide status whether the helm background is ready or not.

```
$ vela traits
NAME           STATUS  DESCRIPTION
autoscale    Ready     Automatically scale workloads by cron or resource utilization
```

File: docs/developers/references/traits/autoscale.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  App helloworld deployed
+  ```
+  
+  Check the replicas of Deployment `frontend` which is deployed by workload webservice `helloworld` and there is one replica.
+  
+  (TODO: The command below needs to be replaced  with `vela show` to check the replicas.)
+  ```
+  $ kubectl get deploy frontend
+  NAME       READY   UP-TO-DATE   AVAILABLE   AGE
+  frontend   1/1     1            1           2m52s
```
Comment (wonderflow) [inline docs/developers/references/traits/autoscale.md:35]: yes，it's a MUST have, before kubecon release. We should print the replica using `vela status` not `vela show`

File: docs/developers/references/traits/autoscale.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            days=Monday, Tuesday
+            name=cron-test
+            type=cron
+            duration=2h
+  ```
+  
+  The time is `21:07` which is in the active period of the trait which started at `21:00` and the duration is two hours.
+  Check the replicas of Deployment `frontend` again, it has been scaled to 2.
+  ```
+  $ kubectl get deploy
```
Comment (wonderflow) [inline docs/developers/references/traits/autoscale.md:66]: I hope we can use `vela status` to see this result.

File: docs/developers/references/traits/autoscale.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    keda-operator-metrics-apiserver-77598644dd-w7th5   1/1     Running            0          22m
+    ```
+    
+    Try to fix the issue, and those Pods will become `READY`.
+    ```
+    k get pods -n keda
+    NAME                                               READY   STATUS    RESTARTS   AGE
+    keda-operator-695d978ddb-w5qct                     1/1     Running   0          7m53s
+    keda-operator-metrics-apiserver-77598644dd-f4fjq   1/1     Running   0          7m53s
+    ```
```
Comment (wonderflow) [inline docs/developers/references/traits/autoscale.md:100]: Don't add the debug information here, please remove them.

Remember who are reading your docs.

**Review by wonderflow — CHANGES_REQUESTED** [review]

File: charts/vela-core/templates/defwithtemplate/autoscale.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                      timezone: parameter.timezone
+                  }
+              }, ...]
+          }
+        }
+      parameter: {
+      	minReplicas: *1 | int
+      	maxReplicas: *4 | int
+      	name:        *"" | string
+      	type:        *"cron" | string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/autoscale.yaml:43]:
```suggestion
      	type:        "cron"
```

I suggest we naming this traitdefinition by `cronscale` and directly write `cron` as type here.

File: docs/developers/references/traits/autoscale.md (hunk 5 above)
Comment (zzxwill) [reply docs/developers/references/traits/autoscale.md]: +1, thanks.

File: docs/developers/references/traits/autoscale.md (hunk 7 above)
Comment (zzxwill) [reply docs/developers/references/traits/autoscale.md:66]: +1

File: docs/developers/references/traits/autoscale.md (hunk 8 above)
Comment (zzxwill) [reply docs/developers/references/traits/autoscale.md]: Though I said `This part is only for platform builders. If you are an application developer, just skip it.`, but, as you said, this is for app developer. Let me move it to somewhere for platform builders. I'd like to add some more debug cases as I hit those issues and they waste me much time as a platform builder.

File: docs/developers/references/traits/autoscale.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
+# Automatically scale workload by cron or resource utilization
```
Comment (hongchaodeng) [inline docs/developers/references/traits/autoscale.md:1]: Put this to *docs/developers/set-autoscaling.md* instead.

Comment (zzxwill) [reply docs/developers/references/traits/autoscale.md]: Sure.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**